packages/consumption/src/modules/requests/itemProcessors/proposeAttribute/ProposeAttributeRequestItemProcessor.ts

```diff
@@ -71,19 +73,35 @@ export class ProposeAttributeRequestItemProcessor extends GenericRequestItemProc
         let attribute = parsedParams.attribute;
 
         if (parsedParams.isWithExistingAttribute()) {
-            const localAttribute = await this.consumptionController.attributes.getLocalAttribute(parsedParams.attributeId);
+            const foundAttribute = await this.consumptionController.attributes.getLocalAttribute(parsedParams.attributeId);
 
-            if (!localAttribute) {
+            if (typeof foundAttribute === "undefined") {
                 return ValidationResult.error(TransportCoreErrors.general.recordNotFound(LocalAttribute, requestInfo.id.toString()));
             }
 
-            attribute = localAttribute.content;
+            const latestSharedVersion = await this.consumptionController.attributes.getSharedVersionsOfRepositoryAttribute(parsedParams.attributeId, [requestInfo.peer], true);
+            if (latestSharedVersion.length > 0) {
+                if (typeof latestSharedVersion[0].shareInfo?.sourceAttribute === "undefined") {
+                    throw new Error(`The Attribute ${latestSharedVersion[0].id} does not fulfill the requirements of an own shared Attribute.`);
+                }
+
+                const latestSharedVersionSourceAttribute = await this.consumptionController.attributes.getLocalAttribute(latestSharedVersion[0].shareInfo.sourceAttribute);
+                if (typeof latestSharedVersionSourceAttribute === "undefined") {
+                    throw new Error(`The Attribute ${latestSharedVersion[0].shareInfo.sourceAttribute} was not found.`);
+                }
+
+                if (await this.consumptionController.attributes.isSubsequentInSuccession(foundAttribute, latestSharedVersionSourceAttribute)) {
```

Comment:
What if the proposed e.g. "foundAttribute" is the "latestSharedVersionSourceAttribute"? I belive "isSubsequentInSuccession" returns false hence not throwing an error.